src/platform/console_logging.h

```diff
@@ -162,6 +162,21 @@ static void logStatusToConsole(const CHAR16* message, const EFI_STATUS status, c
     logToConsole(::message);
 }
 
+static void logStatusAndMemInfoToConsole(const CHAR16* newMessage, const EFI_STATUS status, const unsigned int lineNumber, const unsigned long long requestedSize)
+{
+    setText(::message, newMessage);
+    appendText(::message, L" (");
+    appendErrorStatus(::message, status);
+    appendText(::message, L") near line ");
+    appendNumber(::message, lineNumber, FALSE);
+    appendText(::message, L"! Requested size: ");
+    appendNumber(::message, requestedSize, FALSE);
+    appendText(::message, L" bytes. Free consecutive block of RAM: ");
+    appendNumber(::message, GetLargestFreeConsecutiveMemory(), FALSE);

```

Comment:
Please add `#include "memory.h"`at the top of the file to make use `GetLargestFreeConsecutiveMemory()` is available